### ISSUE TITLE
Add arithmetic for `arr` namespaces

### ISSUE THREAD

**Issue by mcrumiller** [pola-rs/polars#14541]
### Description

Arrays are equal length series and we should be able to add, subtract, multiply, and divide them if they have the same dtype:

```python
import polars as pl

df = pl.DataFrame({
    "a": pl.Series([[1, 2], [3, 4]], dtype=pl.Array(pl.UInt8, 2)),
    "b": pl.Series([[1, 2], [3, 4]], dtype=pl.Array(pl.UInt8, 2)),
})

# any of these fail, seems reasonable they should be implemented
df["a"] + df["b"]
df.select(pl.col("a") + pl.col("b"))
df.select(pl.sum_horizontal("a", "b"))
```
```
pyo3_runtime.PanicException: `add` operation not supported for dtype `array[u8, 2]`
```

**Comment by ritchie46** [comment]
Yes, we should support this.

**Comment by itamarst** [comment]
These all work now. However, division is broken because the Python code for division tries to cast to `Float64`, when it should really be casting in this case to `pl.Array(pl.Float64, 2)`.

**Comment by itamarst** [comment]
I will probably fix the division issue as part of #9188, where I have the same problem with `List`.
